src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -166,7 +175,16 @@ def _run_binsearch_scaling(
     count = 0
     while True:
         garbage_collection_cuda()
-        trainer.fit_loop.global_step = 0  # reset after each try
+
+        # reset after each try
+        if model.automatic_optimization:
+            optim_progress = trainer.fit_loop.epoch_loop.batch_loop.optimizer_loop.optim_progress
+        else:
+            optim_progress = trainer.fit_loop.epoch_loop.batch_loop.manual_loop.optim_step_progress
+
+        optim_progress.reset()
```

Comment:
the optim loop should call `reset_on_run()`, and manual loop `reset()`. So it will need to be done inside the `if`

```suggestion
        if model.automatic_optimization:
            trainer.fit_loop.epoch_loop.batch_loop.optimizer_loop.optim_progress.reset_on_run()
        else:
            trainer.fit_loop.epoch_loop.batch_loop.manual_loop.optim_step_progress.reset()```

Reply:
shouldn't both call `reset`? we need to make sure it's a fresh start when running a new batch.